src/test/TxTests.cpp

```diff
@@ -516,7 +516,9 @@ getAccountSigners(PublicKey const& k, Application& app)
 TransactionFramePtr
 transactionFromOperationsV0(Application& app, SecretKey const& from,
                             SequenceNumber seq,
-                            const std::vector<Operation>& ops, int fee)
+                            const std::vector<Operation>& ops, int fee,
+                            std::string memoText, bool activateTimeBounds)
```

Comment:
this change doesn't look necessary: you're touching a helper function that is used in many places and it does some weird stuff like not really setting timebounds that seems to be only useful for loadgen.
You should probably just craft transactions the way you want it in loadgen code

(Also you're passing strings by value which is bad)

Reply:
Reverted the changes to `transactionFromOperations`'s and added code to craft transactions in `LoadGenerator.cpp`

Reply:
I'm not a fan of this approach -- LoadGenerator is manually crafting transactions, and  now it has to handle different transactions types depending on protocol versions, which is bad for several reasons. First of all, there's a lot of duplicate code in loadgen and TxTests now. Second, it is not LoadGenerator's responsibility to know how to do craft these transactions. That's why in my original suggestion I was saying that we should modify the existing helper functions in TxTests to do what we want. I do not see a problem with these functions being used in a lot of places, as long as we make sure we don't set those additional parameters where we don't need to (and by using defaults we do not need to modify the other callsites). Wrt not setting the timebounds correctly, that's a bug in the implementation and can be reworked to set the parameters properly (the function signature can also be reworked to pass Memo instead of a string). What do you think @MonsieurNicolas @hidenori-shinohara

Reply:
I'm good with this approach. It does have some duplicate (30~35 lines?), but I like that it's simple and self-contained.

Reply:
I was not suggesting doing protocol specific stuff in loadgen (I agree, this doesn't make sense):

if you look at how test code is normally written, we do things like
```
                        txFrame = root.tx(
                            {payment(a1.getPublicKey(), paymentAmount)});

                        setMinTime(txFrame, 1000);
                        getSignatures(txFrame).clear();
                        txFrame->addSignature(root);
```

I would expect the loadgen code to so something similar (ie: construct a tx from ops, adjust memo, etc).

Reply:
The updated version still does protocol-specific stuff, which is what we want to avoid here. There are helper functions in TransactionBridge which you can use (setMinTime, setMaxTime). You will need to add `setMemo` in TransactionBridge as well.

Reply:
also, there's still a bug I mentioned earlier with how the timebounds are set, or rather not set. Looking into it more, there seems to be other invalid fields, like inflation destination. Such a transaction is invalid, and will fail on application. Can you look into it @hidenori-shinohara ?

Reply:
I added `setMemo` to `TransactionBridge` and used that, `setMinTime` and `setMaxTime` in `LoadGeneration.cpp`. This simplified the code quite a bit!

How can I check the validity of a transaction? The test included in this PR passes and when I run `pretend` mode locally with something like `stellar-core http-command "generateload?mode=pretend&txs=250&txrate=10"`, loadgen runs without any issue.  @marta-lokhova

Reply:
What I mean is that even though the transaction gets accepted for consensus, it fails during application time. You can run stellar-core with —ll trace and inspect the logs during application of pretend transactions, just look for “Operation result”

Reply:
Thanks! I was able to see that it failed, so I updated the code. It now shows

```
2021-07-13T13:58:56.578 GCTI6 [Tx TRACE] {                                                                                             
    "Operation": {                                                                                                                     
        "sourceAccount": null,                                                                                                         
        "body": {                                                                                                                      
            "type": "SET_OPTIONS",                                                                                                     
            "setOptionsOp": {                                                                                                          
                "inflationDest": "GA37S7VX4U4FOBB36SEARTQYY5SJ5GPNSMXMPGTWYG3JNQ5EHGHUVTQU",                                           
                "clearFlags": null,                                                                                                    
                "setFlags": null,                                                                                                      
                "masterWeight": null,                                                                                                  
                "lowThreshold": null,                                                                                                  
                "medThreshold": null,                                                                                                  
                "highThreshold": null,                                                                                                 
                "homeDomain": "************************",                                                                              
                "signer": {                                                                                                            
                    "key": {                                                                                                           
                        "type": "SIGNER_KEY_TYPE_ED25519",                                                                             
                        "ed25519": "0000000000000000000000000000000000000000000000000000000000000000"                                  
                    },                                                                                                                 
                    "weight": 0                                                                                                        
                }                                                                                                                      
            }                                                                                                                          
        }                                                                                                                              
    }                                                                                                                                  
}                                                                                                                                      
2021-07-13T13:58:56.578 GCTI6 [Tx TRACE] {                                                                                             
    "OperationResult": {                                                                                                               
        "code": "opINNER",                                                                                                             
        "tr": {                                                                                                                                                                                                                                                               
            "type": "SET_OPTIONS",
            "setOptionsResult": {
                "code": "SET_OPTIONS_SUCCESS"
            }                                                      
        }                                                          
    }                                                              
}  
```